ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -501,6 +503,38 @@ static int HandleVnodeOperation(
                     goto CleanupAndReturn;
                 }
             }
+            
+            if (ActionBitIsSet(action, KAUTH_VNODE_WRITE_DATA))
+            {
+                if (!TryGetVirtualizationRoot(&perfTracer, context, currentVnode, pid, &root, &vnodeFsidInode, &kauthResult, kauthError))
+                {
+                    goto CleanupAndReturn;
+                }
+                
+                PrjFSFileXAttrData rootXattr = {};
+                SizeOrError xattrResult = Vnode_ReadXattr(currentVnode, PrjFSFileXAttrName, &rootXattr, sizeof(rootXattr));
+                if (xattrResult.error == ENOATTR)
+                {
+                    // Only notify if the file is still a placeholder
+                    goto CleanupAndReturn;
+                }
+                
+                PerfSample enumerateDirectorySample(&perfTracer, PrjFSPerfCounter_VnodeOp_PreConvertToFull);
```

Comment:
Nit-picking: `enumerateDirectorySample` is probably a copy-pasted name and not really appropriate here.

Reply:
Exellent nit!  The place I copied it from does use that var name and honestly I don't think it makes sense their either :-).  I'll wait for clarification on the states above update accordingly.